sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```
@@ -381,7 +381,7 @@ case class TransformWithStateExec(
 
     // If timeout is based on event time, then filter late data based on watermark
     val filteredIter = watermarkPredicateForDataForLateEvents match {
-      case Some(predicate) =>
+      case Some(predicate) if timeMode == TimeMode.EventTime() =>
```

Do we want to throw an error or log something letting the user know about this behavior? Or do you think it's fine?

---

I think its fine. We will note this in the docs as well